.github/workflows/tests.yml

```diff
@@ -38,13 +38,23 @@ jobs:
             PIP_SELECTOR: '[all, tests, coverage]'
             PYTEST_ARGS_COVERAGE: --cov=. --cov-report=xml
             LABEL: -coverage
-          # Run test suite against upstream development version
+          # Run test suite against dependencies development version
           - os: ubuntu
             PYTHON_VERSION: 3.8
             PIP_SELECTOR: '[all, tests]'
-            LABEL: -upstream_dev
-            UPSTREAM_DEV: true
+            LABEL: -dependencies_dev
+            DEPENDENCIES_DEV: true
             DEPENDENCIES: numpy scipy scikit-learn scikit-image
+          - os: ubuntu
+            PYTHON_VERSION: 3.8
+            PIP_SELECTOR: '[all, tests]'
+            LABEL: -dependencies_pre_release
+            PIP_ARGS: '--upgrade --pre'
+          - os: ubuntu
+            PYTHON_VERSION: 3.9
+            PIP_SELECTOR: '[all, tests]'
+            LABEL: -dependencies_pre_release
+            PIP_ARGS: '--upgrade --pre'
```

Comment:
Once numba 0.53 is released, this needs to be removed.

Reply:
Why is it also in lines 51&52 then?

Reply:
I have moved removed these build to https://github.com/hyperspy/hyperspy-extensions-list/blob/ee1a24abc213c270c2e6c1f9bfcea694f0eeeab1/.github/workflows/tests.yml#L27-L38. Initially, to test python 3.9, I needed the pre-release version of numba from pypi and when doing this, I realised that now quite a few libraries are doing some pre-release which is useful to identify regression and report it so that it get sorted for the release.